ext/luawrapper/include/LuaContext.hpp

```diff
@@ -2395,6 +2416,19 @@ struct LuaContext::Pusher<std::tuple<TTypes...>> {
 /**************************************************/
 // specializations of the Reader structures
 
+// opaque Lua references
+template<>
+struct LuaContext::Reader<LuaContext::LuaObject>
+{
+    static auto read(lua_State* state, int index)
+        -> boost::optional<LuaContext::LuaObject>
+    {
+        LuaContext::LuaObject obj;
+        obj.objectInRegistry = std::shared_ptr<LuaContext::ValueInRegistry>(new LuaContext::ValueInRegistry(state, index));
```

Comment:
Just a nit, `std::make_shared<LuaContext::ValueInRegistry>(state, index)` would be a bit better (one allocation instead of two).